### ISSUE TITLE
Getting `XMLSchemaValueError: argument is not a string: None`

### ISSUE THREAD

**Issue by diogojapinto** [sissaschool/xmlschema#189]
Hello,

While attempting to build a software product to interact with "old" governmental platforms, I came across this awesome library. Nonetheless, I'm facing some difficulties.

When running the following 
```python
import xmlschema
schema = xmlschema.XMLSchema('https://www.fincen.gov/sites/default/files/schema/base/EFL_TVCXBatchSchema.xsd')
```

it fails with the following stack trace:

```python
---------------------------------------------------------------------------
XMLSchemaValueError                       Traceback (most recent call last)
<ipython-input-19-d90960499a8a> in <module>
----> 1 schema = xmlschema.XMLSchema('https://www.fincen.gov/sites/default/files/schema/base/EFL_TVCXBatchSchema.xsd')

~/anaconda3/envs/dash/lib/python3.7/site-packages/xmlschema/validators/schema.py in __init__(self, source, namespace, validation, global_maps, converter, locations, base_url, defuse, timeout, build, use_meta, loglevel)
    439         try:
    440             if build:
--> 441                 self.maps.build()
    442         finally:
    443             if loglevel is not None:

~/anaconda3/envs/dash/lib/python3.7/site-packages/xmlschema/validators/global_maps.py in build(self)
    607                 }
    608 
--> 609         self.check(filter(lambda x: x.meta_schema is not None, not_built_schemas), self.validation)
    610 
    611     def check(self, schemas=None, validation='strict'):

~/anaconda3/envs/dash/lib/python3.7/site-packages/xmlschema/validators/global_maps.py in check(self, schemas, validation)
    649                     base_type = xsd_type.base_type
    650                     if base_type and base_type.name != XSD_ANY_TYPE and base_type.is_complex():
--> 651                         if not xsd_type.content_type.is_restriction(base_type.content_type):
    652                             xsd_type.parse_error("the derived group is an illegal restriction "
    653                                                  "of the base type group", validation=validation)

~/anaconda3/envs/dash/lib/python3.7/site-packages/xmlschema/validators/groups.py in is_restriction(self, other, check_occurs)
    334             return False
    335         elif self.model == other.model or other.model == 'sequence':
--> 336             return self.is_sequence_restriction(other)
    337         elif other.model == 'all':
    338             return self.is_all_restriction(other)

~/anaconda3/envs/dash/lib/python3.7/site-packages/xmlschema/validators/groups.py in is_sequence_restriction(self, other)
    391                 except StopIteration:
    392                     return False
--> 393                 if other_item is item or item.is_restriction(other_item, check_occurs):
    394                     break
    395                 elif check_emptiable and not other_item.is_emptiable():

~/anaconda3/envs/dash/lib/python3.7/site-packages/xmlschema/validators/elements.py in is_restriction(self, other, check_occurs)
    927                 return False
    928             elif self.fixed != other.fixed and \
--> 929                     raise XMLSchemaAttributeError(str(self.fixed) + ' - ' + XMLSchemaAttributeError(other.fixed))
    930                     self.type.normalize(self.fixed) != other.type.normalize(other.fixed):
    931                 return False

~/anaconda3/envs/dash/lib/python3.7/site-packages/xmlschema/validators/simple_types.py in normalize(self, text)
    365             text = text.decode('utf-8')
    366         elif not isinstance(text, str):
--> 367             raise XMLSchemaValueError('argument is not a string: %r' % text)
    368 
    369         if self.white_space == 'replace':

XMLSchemaValueError: argument is not a string: None
```

Apparently there is a variable (I believe `self.fixed` on `elements.py`, but not sure), that is passed on to normalize despite being set to `None`.

Could you please help debugging this issue?

Version:
```python
Python 3.7.0 | packaged by conda-forge | (default, Nov 12 2018, 12:34:36)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import xmlschema
>>> xmlschema.__version__
'1.1.3'
```

**Comment by brunato** [comment]
Hi,

thank you for reporting this problem. This is related to element restriction checking.  There is not a workaround to avoid the _ValueError_ in v1.1.3. 

I added a patch for v1.2.0 in my develop. This patch replaces lines 928-931 of `xmlschema/validators/elements.py` with:

```python
            elif other.fixed is not None and (
                    self.fixed is None or
                    self.type.normalize(self.fixed) != other.type.normalize(other.fixed)):
                return False
```

**Comment by brunato** [comment]
Fix available with release v1.2.0.
